### PR TITLE
Switch macOS installers to use packages and add vendor concept

### PR DESCRIPTION
**Pull request by gdams**
pkgbuild has been deprecated for quite some time so this PR swaps us to use Packages (same as Corretto)

### CODE REVIEW

File: pkgbuild/Resources/preinstall [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (aahlenst) [inline pkgbuild/Resources/preinstall:1]: /usr/bin/env bash if the bash is somewhere else.

File: pkgbuild/Resources/preinstall [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+JDK_INSTALL_PATH="/Library/Java/JavaVirtualMachines"
+MKDIR=`which mkdir`
```
Comment (aahlenst) [inline pkgbuild/Resources/preinstall:3]: Why this indirection?

File: pkgbuild/packagesbuild.sh [hunk 3]
```diff
@@ -65,42 +98,36 @@ while test $# -gt 0; do
...
-esac
+fi
 
-# Plist commands:
 case $JVM in
   openj9)
-    IDENTIFIER="net.adoptopenjdk.${MAJOR_VERSION}-openj9.${TYPE}"
-    DIRECTORY="adoptopenjdk-${MAJOR_VERSION}-openj9.${TYPE}"
-    BUNDLE="AdoptOpenJDK (OpenJ9)"
+    IDENTIFIER="net.${VENDOR}.${MAJOR_VERSION}-openj9.${TYPE}"
```
Comment (aahlenst) [inline pkgbuild/packagesbuild.sh:115]: Ideally, it should be possible to redeclare the entire identifier.

File: pkgbuild/packagesbuild.sh [hunk 4]
```diff
@@ -65,42 +98,36 @@ while test $# -gt 0; do
...
-      jre) BUNDLE="AdoptOpenJDK (OpenJ9, JRE)" ;;
-      jdk) BUNDLE="AdoptOpenJDK (OpenJ9)" ;;
+      jre) BUNDLE="${PACKAGE_NAME} (OpenJ9, JRE)" ;;
+      jdk) BUNDLE="${PACKAGE_NAME} (OpenJ9)" ;;
     esac
     ;;
   *)
-    IDENTIFIER="net.adoptopenjdk.${MAJOR_VERSION}.${TYPE}"
-    DIRECTORY="adoptopenjdk-${MAJOR_VERSION}.${TYPE}"
+    IDENTIFIER="net.${VENDOR}.${MAJOR_VERSION}.${TYPE}"
```
Comment (aahlenst) [inline pkgbuild/packagesbuild.sh:125]: Ideally, it should be possible to redeclare the entire identifier.

File: pkgbuild/Resources/preinstall (hunk 2 above)
Comment (gdams) [reply pkgbuild/Resources/preinstall:3]: I've copied the Oracle and Corretto preinstall script here, I don't think that's strictly necessary but it shouldn't do any harm

File: pkgbuild/Resources/preinstall (hunk 1 above)
Comment (gdams) [reply pkgbuild/Resources/preinstall:1]: updated

File: pkgbuild/packagesbuild.sh (hunk 3 above)
Comment (gdams) [reply pkgbuild/packagesbuild.sh:115]: add `--identifier`

File: pkgbuild/packagesbuild.sh (hunk 4 above)
Comment (gdams) [reply pkgbuild/packagesbuild.sh:125]: add `--identifier`

File: pkgbuild/packagesbuild.sh [hunk 5]
```diff
@@ -109,35 +147,51 @@ esac
...
-  >distribution.xml ; \
+OUTPUT_FILE=$(basename "$OUTPUT_DIRECTORY" | cut -f 1 -d '.')
+
+rm -rf *.pkg build/*.pkg distribution.xml Resources/en.lproj/welcome.html Resources/en.lproj/conclusion.html OpenJDKPKG.pkgproj "${DIRECTORY}"
+
+cp -R "${INPUT_DIRECTORY}" "${DIRECTORY}"
+
+if [ ! -z "$SIGN_OPTION" ]; then
+    xattr -cr .
+    /usr/bin/codesign --verbose=4 --deep --force -s "Developer ID Application: London Jamocha Community CIC" ${DIRECTORY}
```
Comment (karianna) [inline pkgbuild/packagesbuild.sh:164]: Does this need to be parameterized?

**Review by karianna — CHANGES_REQUESTED** [review]

File: pkgbuild/packagesbuild.sh (hunk 5 above)
Comment (gdams) [reply pkgbuild/packagesbuild.sh:164]: not really, this is Adopt specific code

**Review by karianna — APPROVED** [review]

File: pkgbuild/README.md [hunk 6]
```diff
@@ -1,9 +1,18 @@
...
+## Prerequisites
+
+The generation of the OpenJDK installer depends on an open-source utility tool [Packages](http://s.sudre.free.fr/Software/Packages/about.html).
+The application must be installed before running the installer generation task.
+
+```bash
+brew cask install packages
+```
+
 ## Example Usage
```
Comment (aahlenst) [inline pkgbuild/README.md:12]: Should explain how to override the vendor and that one might shoot themselves in the foot by only overriding vendor and not identifier, too. That could result in a identifier like `net.acme` even if you don't own `net.acme`.

**Review by aahlenst — APPROVED** [review]

**Pull request merged**